public/pages/Snapshots/components/RestoreActivitiesPanel/RestoreActivitiesPanel.tsx

```diff
@@ -102,33 +144,48 @@ export const RestoreActivitiesPanel = ({ snapshotManagementService, snapshotId,
       ) {
         const info = response[responseItem].shards[0];
         const stage = stages.indexOf(info.stage);
-        const size = `${(info.index.size.total_in_bytes / 1024 ** 2).toFixed(2)}mb`;
-
         const time = {
           start_time: info.start_time_in_millis,
           stop_time: info.stop_time_in_millis ? info.stop_time_in_millis : Date.now()
         };
 
-        doneCount = stage === 4 ? doneCount + 1 : doneCount;
+        doneCount = stage === lastStage ? doneCount + 1 : doneCount;
         stageIndex = stage < stageIndex ? stage : stageIndex;
 
         maxStopTime = maxStopTime && maxStopTime > time.stop_time ? maxStopTime : time.stop_time;
 
+        const indexStatus = modifiedStages[info.stage as keyof ModifiedStages];
+
         if (info.source.index && info.source.snapshot === snapshotId) {
           minStartTime = minStartTime && minStartTime < time.start_time ? minStartTime : time.start_time;
-          indexes.push({ index: info.source.index, "store.size": size });
+          indexes.push({ index: info.source.index, "restore_status": indexStatus });
         }
       }
     }
-    let percent = Math.floor((doneCount / restoreCount) * 100);
 
-    setIndices(indexes);
+    const updatedIndices = [...indexes];
+    const indicesStarted = indexes.map((index) => index.index);
+
+    for (let index of indicesToRestore) {
+      if (indicesStarted.indexOf(index) < 0) {
+        updatedIndices.push({ index, restore_status: "Pending" })
+      }
+    }
+
+    const sortedUpdatedIndices = updatedIndices.sort((a, b) => {
+      if (a.index < b.index) return -1;
+      if (a.index > b.index) return 1;
+      return 0;
+    });
+    const percent = Math.floor((doneCount / restoreCount) * 100);
+
+    setIndices(sortedUpdatedIndices);
     setStopTime(new Date(maxStopTime).toLocaleString().replace(",", "  "));
     setStartTime(new Date(minStartTime).toLocaleString().replace(",", "  "))
 
     if (stages[stageIndex]) {
-      stageIndex = (stageIndex === 4 && doneCount < restoreCount) ? 2 : stageIndex;
-      setStage(`${stages[stageIndex][0] + stages[stageIndex].toLowerCase().slice(1)} (${percent}%)`);
+      stageIndex = (stageIndex === lastStage && doneCount < restoreCount) ? 2 : stageIndex;
```

Comment:
Just want some clarity on this line because it's not straightforward what transformation we are doing here
- What does the 2 mean 
- What's the usage of restoreCount

Reply:
The 2 represents the longest stage of a restore operation.  This line is saying that if the stageIndex has reached the final stage but not all indices are restored, set the stage at the most likely stage of any indices remaining to be restored.  I put this here after encountering such a situation, but I will test without it and remove it if it has no effect.   Restore count represents the total number of indices from the snapshot that have been selected to restore.